getting_started_modules.md

```diff
@@ -0,0 +1,167 @@
+
+> :warning: Read the section `Description - what & why` in [README.md](/README.md) to get a quick introduction to what this repo is.
+
+The template you found this README in is specifically built to make it as easy and quick as possible to make terraform modules, and then test them inside the [vagrant-hashistack](https://app.vagrantup.com/fredrikhgrelland/boxes/hashistack) box. 
+
+This guide aims to show you how to use this template. The steps we are going to walk through are as follows: TODO:UPDATE THIS LIST
+
+- Building a docker image
+- Creating a nomad job that uses this image
+- Creating the terraform module
+- Making the nomad job more dynamic
+
+
+### The development process
+
+#### 1. Building Docker Image
+
+> :warning: This section is only relevant if you want to build your own docker image. Skip this if you are new.
+
+Most of the terraform modules will deploy one or more docker-containers to Nomad. If you want to create your own docker-image, put the [`Dockerfile`](https://docs.docker.com/engine/reference/builder/) under [docker/](/docker/).
+
+#### 2. Deploying Container With Nomad
+At this point you should have a service that will run when you start the docker container. TODO:ADD LINK TO SIMPLE CONTAINER TO USE FOR GUIDE Either you've made a container yourself, or you are using some other container. The next step is then to deploy this container to our hashistack ecosystem. Nomad is running inside our virtual machine, and is used to deploy containers, and register them into Consul. It also has a tight integration with vault that we will use later. 
+
+##### a. Making image available to Nomad
+
+> :warning: Skip this step if you are using a pre-made container
+
+The image we built in our first step is now available as an image on our local machine, but our virtual machine does not have access to that. We need to take our docker image and make that available inside our box (because that is where the nomad server is running). To be able to transfer files from our local machine to the vagrant box we are going to use MinIO which was mentioned earlier. As mentioned in [Making artifacts available](#2-minio) everything put in the root directory will be available in MinIO. [This section](#pushing-resources-to-minio-with-ansible-docker-image) shows how we can use ansible code to:
+
+- 1. create a tmp folder 
+- 2. build and archive our docker image in that tmp folder 
+
+##### b. Creating a nomad job
+Next step is to create the nomad job that deploys our image. This guide will not focus on how to make a nomad job, but a full example can be found at [template_example/conf/nomad/countdash.hcl](template_example/conf/nomad/countdash.hcl). Your nomad job-file should go under `conf/nomad/`. If you made your own docker image see [fetching docker image](#fetching-resources-from-minio-with-nomad-docker-image) on how to use that in your nomad job. When the nomad job-file has been created we can try and run it. We can do this one of two ways:
```

Comment:
Where is this section: `fetching-resources-from-minio-with-nomad-docker-image`?

Reply:
Missing link, will fix!